docs/ActiveLabel.rst

```diff
@@ -1,135 +1,320 @@
 
 .. code-block:: ruby
 
-  Person.labeled_as_destroyed
+  Person.actor.all
+
+  Person.actor.first
 
-  Person.first.labeled_as_destroyed?
+Calling ``Person.actor`` simply adds a label scope, which can be combined with any custom scopes you have (e.g.
+``Person.most_popular`` -> ``Person.actor.most_popular``
 
 Associations
 ~~~~~~~~~~~~
 
-You can even create associations to traverse to labels:
+You can create associations with ActiveLabels:
 
 .. code-block:: ruby
 
-  class Organization
+  class Movie
     include Neo4j::ActiveNode
 
-    has_many :out, :addressables, type: :HAS_ADDRESSABLE_OBJECT, label_class: :HasAddress
+    has_many :in, :actors, type: :ACTS_IN, label_module: :Actor
 
-    # `model_class` acts as a filter to the `label_class` argument.  Both `model_class` and `label_class` can be arrays
-    has_many :out, :addressable_people, type: :HAS_ADDRESSABLE_OBJECT, label_class: :HasAddress, model_class: :Person
+    # `label_module` acts as a filter to the `model_class` argument.
+    # Both `model_class` and `label_module` can be arrays
+    has_many :in, :human_actors, type: :ACTS_IN, label_module: :Actor, model_class: :Person
   end
 
+If you want more control over your association, you can use the ``node_labels:`` option instead
+
+.. code-block:: ruby
+
+  class Movie
+    include Neo4j::ActiveNode
+
+    # The node_labels option accepts a two dimentional array. Each array in the node_labels
+    # array includes a set of labels that the association will match against. In the example
+    # below, the ``actors`` association only includes nodes which have either ``:Actor:Person``
+    # OR ``:Actor:Animal`` labels and have an ``<-[:ACTS_IN]-`` relation to a ``Movie`` node
+    has_many :in, :actors, type: :ACTS_IN, node_labels: [[:Actor, :Person], [:Actor, :Animal]]
+
+    # Other valid params for the node_labels option are
+    has_many :in, :actors, type: :ACTS_IN, node_labels: [[:Actor, :Person], :Actor]
+
+    # or
+    has_many :in, :actors, type: :ACTS_IN, node_labels: :Actor
+  end
+
+Note, while the ``label_module`` option requires its params to resolve to ``ActiveLabel`` modules, the ``node_labels``
+option doesn't. The ``node_labels`` option simply matches against the specified labels.
+
+Multiple Conditions
+-------------------
+
+Sometimes you may wish for ``ActiveLabel`` code to be associated with an array of labels, rather than a single label.
+Perhaps the code triggers if `any` label in the array is present, or perhaps it only triggers if `all` labels in the
+array are present.
+
+.. code-block:: ruby
+
+  module Hollywood
+    include Neo4j::ActiveLabel
+
+    self.associated_labels = [:Actor, :Director]
+    self.associated_labels_matcher = :any
+
+    # OR
+
+    self.associated_labels = [:Actor, :Director]
+    self.associated_labels_matcher = :all
+    
+  end
+
+By default, ``self.associated_labels_matcher == :any``
```

Comment:
Allowing ActiveLabels to have multiple associated labels would also be the primary way of sharing code between multiple `ActiveLabel` modules, rather than using something like `ActiveSupport::Concern`. I realized that sharing code using a concern wouldn't work, because there isn't a good (any?) way of specifying that the concern's code should only apply to certain instances of a class.